sysinfo/device_type.go

```diff
@@ -0,0 +1,100 @@
+package sysinfo
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+)
+
+type SystemDeviceType string
+
+const (
+	SystemDeviceTypeUnknown SystemDeviceType = "unknown"
+	SystemDeviceTypeDesktop SystemDeviceType = "desktop"
+	SystemDeviceTypeServer  SystemDeviceType = "server"
+)
+
+// detectBySystemDefaultTarget determines the device type based on the default systemd target.
+func detectBySystemDefaultTarget() SystemDeviceType {
+	_, err := exec.LookPath("systemctl")
+	if err != nil {
+		return SystemDeviceTypeUnknown
+	}
+
+	out, err := exec.Command("systemctl", "get-default").Output()
+	if err != nil {
+		return SystemDeviceTypeUnknown
+	}
+
+	switch strings.TrimSpace(string(out)) {
+	case "graphical.target":
+		return SystemDeviceTypeDesktop
+	case "multi-user.target":
+		return SystemDeviceTypeServer
```

Comment:
The default target can be modified, but I guess this is good enough for our case?